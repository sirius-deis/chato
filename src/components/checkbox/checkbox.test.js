import { render, screen, fireEvent } from '@testing-library/react';
import LabelWithCheckbox from './checkbox';
import React from 'react';

const ParentComponent = () => {
  const [isChecked, setIsChecked] = React.useState(false);

  const handleCheckboxChange = () => {
    setIsChecked(prevState => !prevState);
  };

  return (
    <LabelWithCheckbox
      label="Click me"
      isChecked={isChecked}
      onChangeHandler={handleCheckboxChange}
    />
  );
};

describe('Checkbox component', () => {
  it('should match snapshot', () => {
    const { container } = render(<LabelWithCheckbox label='Click me' />);
    expect(container).toMatchSnapshot();
  });
  it('should contain provided text', () => {
    render(<LabelWithCheckbox label='Click me' />);
    expect(screen.getByText(/Click me/)).toBeInTheDocument();
  });
  it('should fire an event when clicking the component', () => {
    render(<ParentComponent />);
    expect(screen.queryByTestId('checkbox').checked).toBeFalsy();
    fireEvent.click(screen.getByText(/Click me/));
    expect(screen.queryByTestId('checkbox').checked).toBeTruthy();
  });
});
