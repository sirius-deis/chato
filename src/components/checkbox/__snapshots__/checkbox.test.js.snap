// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkbox component should match snapshot 1`] = `
<div>
  <label
    class="sc-aXZVg byPRVZ"
  >
    <input
      class="sc-gEvEer PwOzy"
      data-testid="checkbox"
      type="checkbox"
    />
    <span
      class="sc-eqUAAy dzTgSe"
    />
    Click me
  </label>
</div>
`;
